pkg/csi/service/vanilla/controller.go

```diff
@@ -1068,6 +1087,16 @@ func (c *controller) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequ
 					"failed to delete volumeInfo CR for volume: %q. Error: %+v", req.VolumeId, err)
 			}
 		}
+		// If this is multi-VC configuration, delete CnsVolumeInfo CR
+		if multivCenterCSITopologyEnabled && len(c.managers.VcenterConfigs) > 1 {
+			if volumeInfoService != nil {
```

Comment:
we can remove this check `if volumeInfoService != nil {`

Reply:
done